cmake/compiler/icx/generic.cmake

```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if(DEFINED TOOLCHAIN_HOME)
+  set(find_program_icx_args PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+endif()
+
+find_program(CMAKE_C_COMPILER icx ${find_program_icx_args})
+
+find_program(CMAKE_DTS_PREPROCESSOR gcc)
```

Comment:
can't oneapi be used instead of host GCC ?

Reply:
good point, just tried, and it actually works with icx